examples/advanced/advanced.go

```diff
@@ -0,0 +1,182 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"oras.land/oras-go/pkg/content"
+	"oras.land/oras-go/pkg/oras"
+	"oras.land/oras-go/pkg/target"
+)
+
+func main() {
+	var verbose int
+	cmd := &cobra.Command{
+		Use:          fmt.Sprintf("%s [command]", os.Args[0]),
+		SilenceUsage: true,
+		PersistentPreRun: func(cmd *cobra.Command, args []string) {
+			log.SetLevel(log.InfoLevel)
+			if verbose > 1 {
+				log.SetLevel(log.DebugLevel)
+			}
+		},
+	}
+	cmd.AddCommand(copyCmd())
+	cmd.PersistentFlags().IntVarP(&verbose, "verbose", "v", 1, "set log level")
+	if err := cmd.Execute(); err != nil {
+		os.Exit(1)
+	}
+}
+
+func copyCmd() *cobra.Command {
+	var (
+		fromStr, toStr string
+		manifestConfig string
+		opts           content.RegistryOptions
+	)
+	cmd := &cobra.Command{
+		Use:   "copy <name:tag|name@digest>",
+		Short: "Copy artifacts from one location to another",
+		Long: `Copy artifacts from one location to another
+Example - Copy artifacts from local files to local files:
+  oras copy foo/bar:v1 --from files --to files:path/to/save file1 file2 ... filen
+Example - Copy artifacts from registry to local files:
+  oras copy foo/bar:v1 --from registry --to files:path/to/save
+Example - Copy artifacts from registry to oci:
+  oras copy foo/bar:v1 --from registry --to oci:path/to/oci
+Example - Copy artifacts from local files to registry:
+  oras copy foo/bar:v1 --from files --to registry file1 file2 ... filen
+
+When the source (--from) is "files", the config by default will be "{}" and of media type
+application/vnd.unknown.config.v1+json. You can override it by setting the path, for example:
+
+  oras copy foo/bar:v1 --from files --manifest-config path/to/config:application/vnd.oci.image.config.v1+json --to files:path/to/save file1 file2 ... filen
+
+
+`,
+		Args: cobra.MinimumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			var (
+				ref      = args[0]
+				err      error
+				from, to target.Target
+			)
+			// get the fromStr; it might also have a ':' to add options
+			fromParts := strings.SplitN(fromStr, ":", 2)
+			toParts := strings.SplitN(toStr, ":", 2)
+			switch fromParts[0] {
+			case "files":
+				fromFile := content.NewFile("")
```

Comment:
@deitch a question for my understanding - is this content.NewFile (which creates a new File interface) basically replacing / renaming what is currently called a FileStore? -> https://github.com/oras-project/oras-go/blob/0861304ddc43f47299a9c04933d4b4e16adad472/pkg/content/file.go#L42. Are there subtle differences between the two?

Reply:
It is indeed the continuation of it. `FileStore` actually implements `content.Provider` and `content.Ingester`, while `File` implements `remote.Resolver`.